Add "POS" functionality to the $whohas command

ulrikjohansson opened the pull request:
I had some fun over the weekend, and made a stab at making a "POS" filter for the $whohas command.

It works by checking a "POS" column in the offers sheet for a comma separated list of "NaturalIds" (no mapping between Names and Ids for now).
If there exists a POS column, and that column is not empty, we filter FIO inventory data to only count amounts at the listed locations.

Example:
Kindling has 1000 C on EW, and 500 C on Deimos. This is what will get reported from FIO
Kindling does not want the $whohas command to show the Deimos inventory, b.c it's not for sale, it's for smelting.
So they add "KW-688c" to the POS column for their C offfer row.

Now when someone asks $whohas C, it will respond with "KINDLING has 1000 C"

If they would have left the POS column empty, it would have responded with "KINDLING has 1500 C"
If they would have put [KW-688c,ZV-759c] in the POS column, it would also respond with 1500 C

I felt I needed to add some tests in order to not introduce tons of bugs, and to make testing a bit easier, I needed to refactor some of the functions into smaller pieces.
I also added some typing info to the functions I worked with. Also to make it a little easier to deal with all the transformations of the various lists and dicts and tuples we throw around.
I hope this actually makes it easier to continue work on for others as well. If this just made everything more confusing, please point me in the direction you'd like this to go, and I'll refactor towards that.

gilith1 commented on test/test_inventory.py:
```diff
@@ -0,0 +1,129 @@
+    inv = await bot.whohas(MagicMock(), "C", False)
+
+    assert len(inv) == 2
+    assert inv[0] == ("Kindling", 200)
+    assert inv[1] == ("Gilith", 100)
```
maybe change this to:
assert ("Kindling", 200) in inv
we shouldn't rely on order, it might change in the future if we sort the results differently
this also applies to test_inventory

gilith1 commented on HAL9666/AuctionMasterBot.py:
```diff
@@ -560,4 +561,5 @@ async def clearchannel(ctx):
 #keep_alive()
-bot.run(os.getenv('DISCORD_TOKEN'))
+if __name__ == "__main__":
+  bot.run(os.getenv('DISCORD_TOKEN'))
```
it would make sense to move the intents stuff and bot initialization (lines 86-93) down here too

gilith1 reviewed (commented):
Now that findInInventory and updateInventory are no longer async, they should be called directly, without await

ulrikjohansson commented:
Fixed a few things, I must have forgot to actually run the tests after the last fixes!

It's not possible to move the bot initialization stuff down into the if __name__ == "__main__" bit, since `bot` will be undefined when we try to use it for the decorators. So I moved it back for now.
Also, I actually ran the tests this time 😅
